dash/App/Tokens/AddToken/index.js

```diff
@@ -377,43 +429,26 @@ class AddTokenFormScreenComponent extends Component {
 const AddTokenFormScreen = Restore.connect(AddTokenFormScreenComponent)
 
 class AddToken extends Component {
-  constructor(props, context) {
-    super(props, context)
-    this.state = {
-      tokenData: {
-        name: '',
-        symbol: '',
-        decimals: ''
-      }
-    }
-  }
-
-  async updateTokenData(contractAddress, chainId) {
-    const { name, symbol, decimals } = await link.invoke('tray:getTokenDetails', contractAddress, chainId)
-    this.setState({ tokenData: { name, symbol, decimals } })
-  }
-
   render() {
     const { data, req } = this.props
-    const address = data && data.notifyData && data.notifyData.address
-    const chainId = data && data.notifyData && data.notifyData.chainId
+    const { address, chainId, error, tokenData, isEdit } = data?.notifyData || {}
     const chainName = chainId ? this.store('main.networks.ethereum', chainId, 'name') : undefined
 
-    if (!chainId) {
-      return <AddTokenChainScreen />
-    } else if (!address) {
-      return (
-        <AddTokenAddressScreen
-          chainId={chainId}
-          chainName={chainName}
-          updateTokenData={this.updateTokenData.bind(this)}
-        />
-      )
-    }
+    if (!chainId) return <AddTokenChainScreen />
+    if (!address) return <AddTokenAddressScreen chainId={chainId} chainName={chainName} />
 
-    const tokenData = { address, ...this.state.tokenData }
+    //Errors can only occur after the address form has been presented
```

Comment:
can you explain this comment a little more? how does it inform a developer's decisions in the future?

Reply:
@Jamchello just a reminder to remove this comment